errors on building opencv with MinGW on windows without VisualStudio

##### System information (version)

- OpenCV => 4.2
- Operating System / Platform => Windows 10 64 Bit
- Compiler => MinGW64

##### Detailed description

 I am trying build latest version openCV library via MinGW on Window10 64 bit machine without Microsoft Visual Studio. I know that is a little bit tricky method but I don't like use Microsoft build tools.  I handled the errors I have encountered so far except latest one.


>In addition, I checked out [Microsoft VS Docs](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/isnan-isnan-isnanf?view=msvc-160). That line confused me : " **The _isnan and _isnanf functions are Microsoft-specific.** ".  I'm not sure if the method I use is wrong for my operating system. I also checked out openCV Docs and FAQ, I didn't find any solution for my problem. 

The error I get is below :

[ 45%] Building CXX object modules/wechat_qrcode/CMakeFiles/opencv_wechat_qrcode.dir/src/binarizermgr.cpp.obj
In file included from D:\opencvInstall\test3\opencv_contrib\modules\wechat_qrcode\src\zxing/common/bitarray.hpp:15,
                 from D:\opencvInstall\test3\opencv_contrib\modules\wechat_qrcode\src\zxing/common/bytematrix.hpp:13,
                 from D:\opencvInstall\test3\opencv_contrib\modules\wechat_qrcode\src\imgsource.hpp:13,
                 from D:\opencvInstall\test3\opencv_contrib\modules\wechat_qrcode\src\precomp.hpp:25,
                 from D:\opencvInstall\test3\opencv_contrib\modules\wechat_qrcode\src\binarizermgr.cpp:7:
D:/opencvInstall/test3/opencv_contrib/modules/wechat_qrcode/src/zxing/zxing.hpp: In function 'bool zxing::isnan(float)':
D:/opencvInstall/test3/opencv_contrib/modules/wechat_qrcode/src/zxing/zxing.hpp:64:37: error: '_isnan' was not declared in this scope; did you mean 'isnan'?
   64 | inline bool isnan(float v) { return _isnan(v) != 0; }
   ......|....................................................................^~~~~~
   ......|....................................................................isnan
D:/opencvInstall/test3/opencv_contrib/modules/wechat_qrcode/src/zxing/zxing.hpp: In function 'bool zxing::isnan(double)':
D:/opencvInstall/test3/opencv_contrib/modules/wechat_qrcode/src/zxing/zxing.hpp:65:38: error: '_isnan' was not declared in this scope; did you mean 'isnan'?
   65 | inline bool isnan(double v) { return _isnan(v) != 0; }
   ......|....................................................................^~~~~~
   ......|....................................................................isnan
mingw32-make[2]: *** [modules\wechat_qrcode\CMakeFiles\opencv_wechat_qrcode.dir\build.make:82: modules/wechat_qrcode/CMakeFiles/opencv_wechat_qrcode.dir/src/binarizermgr.cpp.obj] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:7468: modules/wechat_qrcode/CMakeFiles/opencv_wechat_qrcode.dir/all] Error 2



######  That is my first issue report on github. If there is any mistake in format so sorry.

This would appear to not have corrected the issue, or perhaps it exposed a different issue. In the latest release 4.5.2 I receive the following error when building:

```
[ 71%] Linking CXX shared library ..\..\bin\libopencv_wechat_qrcode452.dll
C:/mingw-w64/x86_64-7.3.0-posix-seh-rt_v5-rev0/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/7.3.0/../../../../x86_64-w64-mingw32/lib/../lib/libiconv.a(localcharset.o):localcharset.c:(.text+0x63): undefined reference to `__imp_GetACP'
collect2.exe: error: ld returned 1 exit status
mingw32-make[2]: *** [modules\wechat_qrcode\CMakeFiles\opencv_wechat_qrcode.dir\build.make:976: bin/libopencv_wechat_qrcode452.dll] Error 1
mingw32-make[1]: *** [CMakeFiles\Makefile2:4547: modules/wechat_qrcode/CMakeFiles/opencv_wechat_qrcode.dir/all] Error 2
```

@deadprogram 
i get the same error (`undefined reference to `__imp_GetACP'`) using x86_64-posix-seh-rev0 8.1.0
it seems to need to link to libiconv.a

--- a/modules/wechat_qrcode/src/zxing/zxing.hpp
+++ b/modules/wechat_qrcode/src/zxing/zxing.hpp
@@ -27,7 +27,7 @@
 #define USE_ONED_WRITER 1
 #endif
 
-#if defined(__ANDROID_API__)
+#if defined(__ANDROID_API__) || defined(_MSC_VER) || defined(__MINGW32__) || defined(__MINGW64__)
 
 #ifndef NO_ICONV
 #define NO_ICONV
@@ -51,15 +51,6 @@ typedef unsigned char boolean;
 }  // namespace zxing
 
 #include <limits>
-
-#if defined(_MSC_VER)
-
-#ifndef NO_ICONV
-#define NO_ICONV
-#endif
-
-#endif
-
 #include <cmath>
 
 namespace zxing {

